backend/arm64/emit.ml

```diff
@@ -2127,72 +2154,93 @@ let begin_assembly _unix =
-  emit_printf "\t.data\n";
-  emit_printf "\t.globl\t%a\n" femit_symbol lbl_begin;
-  emit_printf "%a:\n" femit_symbol lbl_begin;
-  let lbl_begin = Cmm_helpers.make_symbol "code_begin" in
-  emit_named_text_section lbl_begin;
-  emit_printf "\t.globl\t%a\n" femit_symbol lbl_begin;
-  emit_printf "%a:\n" femit_symbol lbl_begin;
+  let data_begin = Cmm_helpers.make_symbol "data_begin" in
+  let data_begin_sym = S.create data_begin in
+  D.data ();
+  D.global data_begin_sym;
+  D.define_symbol_label ~section:Data data_begin_sym;
+  let code_begin = Cmm_helpers.make_symbol "code_begin" in
+  let code_begin_sym = S.create code_begin in
+  emit_named_text_section code_begin;
+  D.global code_begin_sym;
+  D.define_symbol_label ~section:Text code_begin_sym;
   (* we need to pad here to avoid collision for the unwind test between the
      code_begin symbol and the first function. (See also #4690) Alignment is
      needed to avoid linker warnings for shared_startup__code_{begin,end} (e.g.
      tests/lib-dynlink-pr4839). *)
   if macosx
   then (
     DSL.ins I.NOP [||];
-    emit_printf "\t.align\t3\n");
-  let lbl_end = Cmm_helpers.make_symbol "code_end" in
-  Emitaux.Dwarf_helpers.begin_dwarf ~code_begin:lbl_begin ~code_end:lbl_end
-    ~file_emitter
+    D.align ~bytes:8);
+  let code_end = Cmm_helpers.make_symbol "code_end" in
+  Emitaux.Dwarf_helpers.begin_dwarf ~code_begin ~code_end ~file_emitter
 
 let end_assembly () =
-  let lbl_end = Cmm_helpers.make_symbol "code_end" in
-  emit_named_text_section lbl_end;
-  emit_printf "\t.globl\t%a\n" femit_symbol lbl_end;
-  emit_printf "%a:\n" femit_symbol lbl_end;
-  let lbl_end = Cmm_helpers.make_symbol "data_end" in
-  emit_printf "\t.data\n";
-  emit_printf "\t.8byte\t0\n";
+  let code_end = Cmm_helpers.make_symbol "code_end" in
+  let code_end_sym = S.create code_end in
+  emit_named_text_section code_end;
+  D.global code_end_sym;
+  D.define_symbol_label ~section:Text code_end_sym;
+  let data_end = Cmm_helpers.make_symbol "data_end" in
+  let data_end_sym = S.create data_end in
+  D.data ();
+  D.int64 0L;
   (* PR#6329 *)
-  emit_printf "\t.globl\t%a\n" femit_symbol lbl_end;
-  emit_printf "%a:\n" femit_symbol lbl_end;
-  emit_printf "\t.8byte\t0\n";
-  emit_printf "\t.align\t3\n";
+  D.global data_end_sym;
+  D.define_symbol_label ~section:Data data_end_sym;
+  D.int64 0L;
+  D.align ~bytes:8;
   (* #7887 *)
-  let lbl = Cmm_helpers.make_symbol "frametable" in
-  emit_printf "\t.globl\t%a\n" femit_symbol lbl;
-  emit_printf "%a:\n" femit_symbol lbl;
+  let frametable = Cmm_helpers.make_symbol "frametable" in
+  let frametable_sym = S.create frametable in
+  D.global frametable_sym;
+  D.define_symbol_label ~section:Data frametable_sym;
   emit_frames
     { efa_code_label =
         (fun lbl ->
-          emit_symbol_type femit_label lbl "function";
-          emit_printf "\t.8byte\t%a\n" femit_label lbl);
+          let lbl = label_to_asm_label ~section:Text lbl in
```

Comment:
I wonder if the Linear language should use Asm_label, which would remove a lot of label_to_asm_label calls (maybe all of them)?  Maybe leave a CR in linear.mli.